client-core/src/cipher/default.rs

```diff
@@ -93,7 +91,7 @@ impl TransactionObfuscation for DefaultTransactionObfuscation {
     fn decrypt(
         &self,
         transaction_ids: &[TxId],
-        private_key: &PrivateKey,
+        private_key_location: &PrivateKeyLocation,
```

Comment:
Does `decrypt` need to take a `PrivateKeyLocation`? if you're implementing the "mixed" mode, then this could just be `PrivateKey`?